addon/mixins/route.js

```diff
@@ -115,7 +125,15 @@ export default Ember.Mixin.create({
   */
   infinityModel(modelName, options) {
 
-    if (Ember.isEmpty(this.store) || Ember.isEmpty(this.store.find)){
+    if (emberDataVersionIs('greaterThan', '1.0.0-beta.19.2') && emberDataVersionIs('lessThan', '1.13.4')) {
```

Comment:
We should outline all of the ember-infinity use cases and ensure that ember-data tests are in place, to protect against future incompatibilities. Skipping many versions of a library can make upgrading apps difficult.